### PR TITLE
Fixes #5522, bug in custom scale example

### LINKED ISSUE #5522: axhline fails on custom scale example
I am using the matplotlib demo [custom_scale_example.py](http://matplotlib.org/1.5.0/examples/api/custom_scale_example.html) as a starting point for my own custom scale.  When I try to add a vertical line or horizontal line to an axis using the demo scaling using [matplotlib.axes.Axes.axhline](http://matplotlib.org/api/axes_api.html?highlight=axhline#matplotlib.axes.Axes.axhline), I get the following error message

    ValueError: It is not possible to compute transA - transB since transB cannot be inverted and there is no shortcut possible.

Here is [gist reproducing this behavior](https://gist.github.com/jeffrey-hokanson/aa21f18825d307221b4a).  

This bug is similar #1246.

### PR DESCRIPTION
**Pull request by jeffrey-hokanson**
Added the property has_inverse = True to the custom scale example so that if axhline is later called on this axis an error is not produced.

### CODE REVIEW

File: examples/api/custom_scale_example.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ class MercatorLatitudeTransform(mtransforms.Transform):
         input_dims = 1
         output_dims = 1
         is_separable = True
-
+        has_inverse = True
```
Comment (mdboom) [inline examples/api/custom_scale_example.py]: Just a PEP8 nitpick here: Insert a blank space before the method definition.

**Comment by mdboom** [comment]
Other than the PEP8 failure, looks good to me.

File: examples/api/custom_scale_example.py [hunk 2]
```diff
@@ -138,6 +138,7 @@ class InvertedMercatorLatitudeTransform(mtransforms.Transform):
         input_dims = 1
         output_dims = 1
         is_separable = True
+        has_inverse = True
```
Comment (jeffrey-hokanson) [inline examples/api/custom_scale_example.py]: I've fixed this and the file appears clear upon running pep8.  Is there something else I'm missing?

**Comment by tacaswell** [comment]
@jeffrey-hokanson Thanks!  Congratulations on (what I think) is your first contribution to mpl!

**Pull request merged**